### PR TITLE
feat(facets): add CRD support in facets and related processing logic

### PR DESCRIPTION
**Pull request by rmvangun**
> [!NOTE]
>
> **Medium Risk**
>
> This PR extends the composer's shared kustomization-synthesis path, which runs for every blueprint that includes a CRD-aware facet. The explicit-wins-implicit contract and path-traversal validation keep blast radius well contained.
>
> **Overview**
>
> The change adds a `crds:` array to the `Facet` type, letting facets declare version-pinned CRD bundle references. The composer deduplicates them across all active facets, synthesizes one `kustomize/crds/<ref>` kustomization per reference with `prune=false` and `wait=true`, and wires every kustomization in the declaring facet to depend on it. An authored kustomization with the same name always takes precedence over synthesis, providing a clean override path. Entries support `${...}` expressions so a single facet can select different CRDs per driver at composition time.
>
> Since the previous review, three follow-up commits closed the two open correctness gaps: `validateCrdRef` now rejects path separators and `..` components, and `addCrdKustomizations` correctly defers to base-blueprint kustomizations. The `strategicMergeKustomization` extension for `Wait`, `Force`, and `Prune` pointer fields and the `resolveSourceName` refactor are also included. Test coverage spans unit, integration, and strategic-merge dimensions across all documented edge cases — including inactive facets, self-dependency cycles, deduplication, and expression-driven driver selection.
>
> Reviewed by Claude for commit `2189844f`.

### CODE REVIEW

File: pkg/composer/blueprint/processor.go [hunk 1]
```diff
@@ -962,6 +971,63 @@ func (p *BaseBlueprintProcessor) collectKustomizations(facet blueprintv1alpha1.F
...
+	for _, ref := range refs {
+		if _, exists := kustomizationByName[ref]; exists {
+			continue
+		}
+		prune := false
+		wait := true
+		kustomizationByName[ref] = &blueprintv1alpha1.ConditionalKustomization{
+			Kustomization: blueprintv1alpha1.Kustomization{
+				Name:   ref,
+				Path:   "crds/" + ref,
```
[the remark this answers is not in the document]
Comment (rmvangun) [reply pkg/composer/blueprint/processor.go:1049]: Addressed in c91cd907. `validateCrdRef` rejects any ref containing `/`, `\`, or `..` in `ProcessFacets` — before the ref is wired into `dependsOn` or concatenated into `crds/<ref>` — so a traversal ref fails composition rather than escaping the catalog dir. Also documented with a schema `pattern: '^[^/\\]+$'`.
